src/main/java/net/dv8tion/jda/core/managers/WebhookManagerUpdatable.java

```diff
@@ -0,0 +1,161 @@
+/*
+ *     Copyright 2015-2016 Austin Keener & Michael Ritter
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.core.managers;
+
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.Icon;
+import net.dv8tion.jda.core.entities.TextChannel;
+import net.dv8tion.jda.core.entities.Webhook;
+import net.dv8tion.jda.core.exceptions.PermissionException;
+import net.dv8tion.jda.core.managers.fields.WebhookField;
+import net.dv8tion.jda.core.requests.Request;
+import net.dv8tion.jda.core.requests.Response;
+import net.dv8tion.jda.core.requests.RestAction;
+import net.dv8tion.jda.core.requests.Route;
+import org.json.JSONObject;
+
+public class WebhookManagerUpdatable
+{
+
+    protected final Webhook webhook;
+
+    protected WebhookField<String> name;
+    protected WebhookField<Icon> avatar;
+    protected WebhookField<TextChannel> channel;
+
+    public WebhookManagerUpdatable(Webhook webhook)
+    {
+        this.webhook = webhook;
+        setupFields();
+    }
+
+    public JDA getJDA()
+    {
+        return webhook.getJDA();
+    }
+
+    public Webhook getWebhook()
+    {
+        return webhook;
+    }
+
+    public WebhookField<String> getNameField()
+    {
+        return name;
+    }
+
+    public WebhookField<Icon> getAvatarField()
+    {
+        return avatar;
+    }
+
+    public WebhookField<TextChannel> getChannelField()
+    {
+        return channel;
+    }
+
+    public void reset()
+    {
+        name.reset();
+        avatar.reset();
+        channel.reset();
+    }
+
+    public RestAction<Void> update()
+    {
+        if (!webhook.getGuild().getSelfMember().hasPermission(webhook.getChannel(), Permission.MANAGE_WEBHOOKS))
+            throw new PermissionException(Permission.MANAGE_WEBHOOKS);
+        if (!shouldUpdate())
+            return new RestAction.EmptyRestAction<>(null);
+
+        JSONObject data = new JSONObject();
+        data.put("name", name.getOriginalValue());
+
+        if (channel.shouldUpdate())
+            data.put("channel_id", channel.getValue().getId());
+        if (name.shouldUpdate())
+            data.put("name", name.getValue());
+        if (avatar.shouldUpdate())
+        {
+            Icon value = avatar.getValue();
```

Comment:
Curious, here you check if avatar's value is null, but in the Field you create, you check for notNull. Is this a mistake?